### PR TITLE
Add Clone Flow to Jetpack Cloud

### PR DESCRIPTION
**Pull request by jwebbdev**
## Proposed Changes

* Implements the clone flow into Jetpack Cloud

- [x] Implement Set Destination Screen
- [x] Implement Select Clone Period Screen
- [x] Implement Configure Screen
- [x] Detect Clone Progress
- [x] Implement Clone Progress Screen
### Known Issues
 * Switching from the Clone flow to Settings retains the state of 'updateRequestStatus' so can show the Finished screen.  Clicking the Finish Up button will return to normal with no ill effects.  The solution here is likely to have the state track a combination of siteId and role for these operations.  Since this isn't breaking and an edge case we're going to follow-up with a fix instead of making this PR larger.

## Testing Instructions

To test cloning you'll need two sites.  A source and a destination.  

The source site needs to have a Jetpack Backup plan and at least one backup.
The destination site needs FTP or SSH/SFTP access, and you'll need the credentials and server info handy.
You can utilize Jurassic Ninja to create both sites if needed.
You may want to make some changes so it'll be obvious the clone took place afterwards.

* Open the Jetpack Cloud Live link
* Select your source site
* Visit the VaultPress backup area
* Add `/clone` to the url - This is temporary until we add links
* Enter your server credentials and test them.  Once they're valid, click 'Finish Up'
* Select the Backup point you wish to clone
* Select the types of items, or leave the default of everything and Confirm
* Wait for the restore to finish and visit the destination site.

### UI Changes to existing pages
* Backup main page has had a header added with a Clone button
* Activity log main page has had the header changed, and includes a Clone button

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

## Screenshots for Translation Context
Screenshot 2023-04-03 at 10 02 37 AM
Screenshot 2023-04-03 at 10 02 46 AM
Screenshot 2023-04-03 at 10 05 53 AM
Screenshot 2023-04-03 at 10 03 23 AM
Screenshot 2023-04-03 at 10 07 01 AM
Screenshot 2023-04-03 at 10 03 37 AM
Screenshot 2023-04-03 at 10 03 43 AM
Screenshot 2023-04-03 at 10 05 29 AM

### CODE REVIEW

**Comment by ederrengifo** [comment]
Few changes to apply based on previous coordination:

- [x] Change the action from "cloning" to "copying" (double check the latest design's copy)
- [x] The call to action must be a secondary button style
- [x] Both call to action in Cloud includes a tooltip when mouseover (check designs)
- [x] Update the navigation bar from the one we already use to get a new license (for partners)
- [x] Standardize the use and positioning of secondary and primary buttons for the steps (aligned to the left)
- [x] Update "point to copy" step (WIP)
- [x] Update views to check the progress of the copy and success scenario (check designs)

File: client/my-sites/activity/activity-log-v2/index.tsx [hunk 1]
```diff
@@ -39,12 +42,28 @@ const ActivityLogV2: FunctionComponent = () => {
...
+							'To test your site changes, migrate or keep your data safe in another site'
+						) }
+					>
+						<Button
+							className="activity-log-v2__clone-button"
+							href={ backupClonePath( selectedSiteSlug ) }
+						>
+							{ translate( 'Copy this site' ) }
+						</Button>
+					</Tooltip>
```
Comment (Initsogar) [inline client/my-sites/activity/activity-log-v2/index.tsx:64]: Not a blocker, but maybe we could consider making this reusable, as it is also required on the backup page and just differs on `className`.

File: client/my-sites/backup/clone-flow/index.tsx [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+import RewindConfigEditor from '../rewind-flow/rewind-config-editor';
+import RewindFlowNotice, { RewindFlowNoticeLevel } from '../rewind-flow/rewind-flow-notice';
+import { defaultRewindConfig, RewindConfig } from '../rewind-flow/types';
+import CloneFlowStepProgress from './step-progress';
+import type { RestoreProgress } from 'calypso/state/data-layer/wpcom/activity-log/rewind/restore-status/type';
+import type { RewindState } from 'calypso/state/data-layer/wpcom/sites/rewind/type';
+import './style.scss';
+
+interface Props {
+	siteId: number;
```
Comment (Initsogar) [inline client/my-sites/backup/clone-flow/index.tsx:45]: Should `siteId` be a required prop? It appears that the majority of selectors and actions depend on it.

File: client/my-sites/backup/clone-flow/index.tsx [hunk 3]
```diff
@@ -0,0 +1,448 @@
...
+	const [ backupPeriod, setBackupPeriod ] = useState< string >( '' );
+	const [ backupDisplayDate, setBackupDisplayDate ] = useState< string >( '' );
+
+	const activityLogPath = '/activity-log/' + siteSlug;
+	const refreshBackups = useCallback(
+		() => dispatch( requestRewindBackups( siteId ) ),
+		[ dispatch, siteId ]
+	);
+	const backupCurrentlyInProgress = useSelector( ( state ) => {
+		if ( ! siteId ) {
```
Comment (Initsogar) [inline client/my-sites/backup/clone-flow/index.tsx:73]: Given that `siteId` is a required prop, is this validation necessary?

File: client/my-sites/backup/clone-flow/index.tsx [hunk 4]
```diff
@@ -0,0 +1,448 @@
...
+				/>
+			</div>
+		</>
+	);
+
+	// Screen that allows user to select a backup point to clone
+	const renderSetBackupPeriod = () => (
+		<>
+			<CloneFlowStepProgress currentStep="clonePoint" />
+			<h3 className="clone-flow__title">{ translate( 'Select a Backup Point to Copy' ) }</h3>
```
Comment (Initsogar) [inline client/my-sites/backup/clone-flow/index.tsx:178]: Maybe we could use caps only for the first word.

```suggestion
			<h3 className="clone-flow__title">{ translate( 'Select a backup point to copy' ) }</h3>
```

File: client/my-sites/backup/clone-flow/test/index.jsx [hunk 5]
```diff
@@ -0,0 +1,468 @@
...
+		},
+		preferences: {
+			remoteValues: {},
+		},
+		documentHead: { unreadCount: 1 },
+	};
+}
+
+function getLogs() {
+	return {
```
Comment (Initsogar) [inline client/my-sites/backup/clone-flow/test/index.jsx:102]: Maybe we could consider moving the fixtures into a separate files like [this example](https://github.com/Automattic/wp-calypso/blob/20fc989ed71ca1b6202043ed0c2a71ace965ed8e/client/reader/discover/test/fixtures/index.js) to make this file smaller.

**Review by Initsogar — APPROVED** [review]
Great work team!

Tested the clone flow using Jurassic Ninja sites from Activity Log and Backup page and works great.

Just left a few comments that are not stoppers, so you could decide to address them in this PR or in a next iteration.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7895803

Thank you @jwebbdev for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7895803).

**Pull request merged**